### PR TITLE
Create generic static logging functions

### PR DESCRIPTION
**Pull request by mborof**
Hey Chris, before I put too much effort into this solution I thought I would validate it will work; I suspect there's a lot more going on there than I realize but basically when that render/show call is made we need a way to get back to event manager which is in whatever frame is running swg-js.  Is that the frame this code is running in?

### CODE REVIEW

**Comment by ChrisAntaki** [comment]
Yep, it's the same frame, thanks for checking!

**Comment by mborof** [comment]
I created standard logging functions so it should be very easy to add logging statements to the gaa code later when we're ready.

File: src/utils/log.js [hunk 1]
```diff
@@ -107,3 +109,50 @@ function toString(val) {
...
+export function logPublisherEvent(event, isFromUserAction, params) {
+  return logAnalyticsEvent(event, EventOriginator.PUBLISHER_CLIENT, isFromUserAction, params);
+}
+
+// Setup code for logging helpers
+let _eventManagerPromiseResolver = null;
+const _eventManagerPromise = new Promise(resolver => _eventManagerPromiseResolver = resolver);
+
+/**
+ * @param {!../api/client-event-manager-api.ClientEventManagerApi} eventManager
```
Comment (ChrisAntaki) [inline src/utils/log.js:141]: Could you describe this method?

File: src/utils/log.js [hunk 2]
```diff
@@ -107,3 +109,50 @@ function toString(val) {
...
+/**
+ * Logs an analytics event
+ * @param {!../proto/api_messages.AnalyticsEvent} eventType
+ * @param {!EventOriginator} eventOriginator 
+ * @param {boolean=} isFromUserAction 
+ * @param {!../proto/api_messages.EventParams=} additionalParameters 
+ * @return {!Promise}
+ */
+async function logAnalyticsEvent(eventType, eventOriginator, isFromUserAction, additionalParameters) {
+  const manager = await _eventManagerPromise;
```
Comment (ChrisAntaki) [inline src/utils/log.js:156]: We can't use async/await in production code yet, unfortunately!

File: src/utils/log.js [hunk 3]
```diff
@@ -107,3 +109,50 @@ function toString(val) {
...
+ * @param {boolean=} isFromUserAction 
+ * @param {!../proto/api_messages.EventParams=} params 
+ * @return {!Promise}
+ */
+export function logPublisherEvent(event, isFromUserAction, params) {
+  return logAnalyticsEvent(event, EventOriginator.PUBLISHER_CLIENT, isFromUserAction, params);
+}
+
+// Setup code for logging helpers
+let _eventManagerPromiseResolver = null;
```
Comment (ChrisAntaki) [inline src/utils/log.js:137]: Could you remove the leading "_" from these local variable names?

**Comment by ChrisAntaki** [comment]
Just curious, were you planning to have swg-gaa.js import `logSwgEvent` and call it?

**Comment by mborof** [comment]
Yes, I'm back out on leave though

**Comment by ChrisAntaki** [comment]
Since `swg.js` and `swg-gaa.js` are different JavaScript binaries, if they both import `logSwgEvent`, they'll end up with different instances of it

**Comment by ChrisAntaki** [comment]
Hey! I'm working on a different approach, will close this PR for now just to clean up the queue

**Pull request closed without merging**